src/components/App.js

```diff
@@ -2,19 +2,32 @@ import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
 import './App.css';
+import calculate from '../logic/calculate';
 
 // eslint-disable-next-line
 class App extends React.Component {
 // eslint-disable-next-line
   constructor(props) {
     super(props);
+    this.state = {
+      next: '',
+      total: '',
+      operation: '',
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    this.setState((prevState) =>
+      Object.assign({}, prevState, calculate(prevState, buttonName))
+    );
   }
 
   render() {
     return (
       <div id="app" className="app-class">
-        <Display />
-        <ButtonPanel />
+        <Display result={this.state.total} />
```

Comment:
destructure total

Reply:
Hey Sunny , how are you ? what you mean with destructure total? In my current code i dont use the other variables so i just play around with total .

Reply:
Hi @userman95 , i'm very well thank you.
```
 const { total } = this.state;
```
line 29 can be
```
 <Display result={total} />
```

Reply:
oh alright ! thanks